pymc_marketing/mmm/tvp.py

```diff
@@ -78,29 +164,54 @@ def time_varying_prior(
         Regression.
     """
 
-    if X_mid is None:
-        X_mid = float(X.mean().eval())
-    if L is None:
-        L = X_mid * 2
-
     model = pm.modelcontext(None)
 
-    if cov_func is None:
-        eta = pm.Exponential(f"{name}_eta", lam=eta_lam)
-        ls = pm.InverseGamma(f"{name}_ls", mu=ls_mu, sigma=ls_sigma)
-        cov_func = eta**2 * pm.gp.cov.Matern52(1, ls=ls)
+    eta = pm.Exponential(f"{name}_eta", lam=-np.log(eta_mass) / eta_upper)
+
+    if ls_upper is None:
+        ls = pc_prior_1d(f"{name}_ls", alpha=1.0 - ls_mass, lower=ls_lower)
+        ls_upper = 2 * X_mid
+    else:
+        # Note: this function sometimes fails, how to handle?
+        # If not using PC prior, should user set ls_mu and ls_sigma instead?
```

Comment:
Do we have any simple heuristics for a reasonable fallback mechanism?

Reply:
None that I know of, I think @ulfaslak told me he wrote something to fix this though?